pandas/_libs/tslibs/offsets.pyx

```diff
@@ -4688,39 +4674,54 @@ cpdef to_offset(freq, bint is_period=False):
 
             tups = zip(split[0::4], split[1::4], split[2::4])
             for n, (sep, stride, name) in enumerate(tups):
-                if is_period is False and name in c_OFFSET_DEPR_FREQSTR:
-                    warnings.warn(
-                        f"\'{name}\' is deprecated, please use "
-                        f"\'{c_OFFSET_DEPR_FREQSTR.get(name)}\' instead.",
-                        FutureWarning,
-                        stacklevel=find_stack_level(),
-                    )
-                    name = c_OFFSET_DEPR_FREQSTR[name]
-                if is_period is True and name in c_REVERSE_OFFSET_DEPR_FREQSTR:
-                    if name.startswith("Y"):
+                if is_period is False and name.upper() in c_OFFSET_DEPR_FREQSTR:
+                    if name == "m" and (n > 0 or stride == "-"):
+                        name = name
+                    else:
```

Comment:
this is super-complicated, I can't understand it anymore

would it work to just keep this part as it is, and at the beginning of `_get_offset`, to add
```python
    if name not in _lite_rule_alias and (name.upper() in _lite_rule_alias):
        warnings.warn(
            f"\'{name}\' is deprecated and will be removed in a future "
            f"version, please use \'{name.upper()}\' instead.",
            FutureWarning,
            stacklevel=find_stack_level(),
        )
    elif name not in _lite_rule_alias and (name.lower() in _lite_rule_alias):
        warnings.warn(
            f"\'{name}\' is deprecated and will be removed in a future "
            f"version, please use \'{name.lower()}\' instead.",
            FutureWarning,
            stacklevel=find_stack_level(),
        )
```
?

Reply:
Thank you for the comment, I added it in `_get_offset()` and we do need this in some cases. But it only works for frequencies in `_lite_rule_alias`. 
The problem I tried to resolve by doing the check
```
if name == "m" and (n > 0 or stride == "-"):
```
was for frequencies like `"2h30m"`. When we pass this freq to `to_offset()` we show our warning: `'m' is deprecated and will be removed in a future version, please use 'ME' instead.`, but we shouldn't.